aiidalab_widgets_base/viewers.py

```diff
@@ -677,16 +682,18 @@ def _default_selection(self):
     def _validate_selection(self, provided):
         return list(provided["value"])
 
-    @observe("selection")
-    def _observe_selection(self, _=None):
-        self.highlight_atoms(self.selection)
-        self._selected_atoms.value = list_to_string_range(self.selection, shift=1)
+    @observe("displayed_selection")
+    def _observe_displayed_selection(self, _=None):
+        self.highlight_atoms(self.displayed_selection)
+        self._selected_atoms.value = list_to_string_range(
+            self.displayed_selection, shift=1
+        )
```

Comment:
I think we agreed that the selected atoms will remain unchanged, as the disappearing formula was very confusing for the users.